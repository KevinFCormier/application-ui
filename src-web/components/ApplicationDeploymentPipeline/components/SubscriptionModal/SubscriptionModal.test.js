/*******************************************************************************
 * Licensed Materials - Property of IBM
 * 5737-E67
 * (c) Copyright IBM Corporation 2016, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/

const React = require('react')
const renderer = require('react-test-renderer')
const SubscriptionModal = require('../SubscriptionModal').default

describe('subscriptionModal', () => {
  it('subscriptionModal renders correctly.', () => {
    const tree = renderer.create(<SubscriptionModal />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
