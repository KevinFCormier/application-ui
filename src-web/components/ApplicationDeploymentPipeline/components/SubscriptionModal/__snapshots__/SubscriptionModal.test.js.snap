// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscriptionModal subscriptionModal renders correctly. 1`] = `
<div
  id="SubscriptionModal"
>
  <div
    className="bx--modal bx--modal-tall modalAddRepo"
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onTransitionEnd={undefined}
    role="presentation"
    tabIndex={-1}
  >
    <div
      aria-label={undefined}
      className="bx--modal-container"
      role="dialog"
    >
      <div
        className="bx--modal-header"
      >
        
        <h2
          className="bx--modal-header__heading"
        >
          
        </h2>
        <button
          className="bx--modal-close"
          onClick={[Function]}
          type="button"
        >
          <svg
            alt="close the modal"
            aria-label="close the modal"
            className="bx--modal-close__icon"
            fill={undefined}
            fillRule="evenodd"
            height="10"
            name={undefined}
            role="img"
            style={undefined}
            viewBox="0 0 10 10"
            width="10"
          >
            <title>
              close the modal
            </title>
            <path
              d="M6.32 5L10 8.68 8.68 10 5 6.32 1.32 10 0 8.68 3.68 5 0 1.32 1.32 0 5 3.68 8.68 0 10 1.32 6.32 5z"
            />
          </svg>
        </button>
      </div>
      <div
        className="bx--modal-content"
      >
        <div
          className="channelGridContainer"
        >
          <div
            className="subscriptionInfoClass"
          >
            <div
              className="subHeader"
            >
              <div
                className="subscriptionInfoHeader"
              >
                Subscription Name
              </div>
              
            </div>
            <div
              className="subHeader"
            >
              <div
                className="subscriptionInfoHeader"
              >
                Namespace
              </div>
              
            </div>
            <div
              className="innerContent"
            >
              <div
                className="placement"
              >
                <div>
                  <div
                    className="label"
                  >
                    Placement
                  </div>
                </div>
                <div>
                  <div
                    className="label-indented"
                  >
                    Clusters
                  </div>
                  <div
                    className="value"
                  >
                    
                  </div>
                </div>
                <div>
                  <div
                    className="label-indented"
                  >
                    Labels
                  </div>
                  <div
                    className="value"
                  >
                     
                    <span
                      className="labelTag"
                      tooltip=""
                    >
                      
                    </span>
                  </div>
                </div>
              </div>
              <div
                className="versions"
              >
                <div>
                  <div
                    className="label"
                  >
                    Deployable versions
                  </div>
                  <div
                    className="value"
                  >
                    version2
                  </div>
                </div>
              </div>
              <div
                className="update"
              >
                <div>
                  <div
                    className="label"
                  >
                    Rolling update
                  </div>
                  <div
                    className="value"
                  >
                    50%
                  </div>
                </div>
              </div>
              <button
                className="editSubscriptionButton"
                onClick={[Function]}
              />
            </div>
          </div>
          <div
            className="channelInfoClass"
          >
            <div
              className="subHeader"
            >
              <div
                className="channelHeader"
              >
                Channel
              </div>
              
            </div>
            <div
              className="innerContent"
            >
              <div
                className="conditions"
              >
                <div>
                  <div
                    className="conditionStatus"
                  >
                    Conditions
                    <span
                      className="conditionStatus"
                    >
                      Conditions fully met
                    </span>
                  </div>
                </div>
                <div
                  className="valueGroup"
                >
                  <div
                    className="valueGroupItem"
                  >
                    <svg
                      alt=""
                      aria-label=""
                      className="icon"
                      fill="#6c7b85"
                      fillRule="evenodd"
                      height="16"
                      name="icon--user"
                      role="img"
                      style={undefined}
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <title>
                        
                      </title>
                      <path
                        d="M4 13.7c1.1.8 2.5 1.3 4 1.3s2.9-.5 4-1.3v-1.2c0-.8-.7-1.5-1.5-1.5h-5c-.8 0-1.5.7-1.5 1.5v1.2zm-1-.8v-.4C3 11.1 4.1 10 5.5 10h5c1.4 0 2.5 1.1 2.5 2.5v.4c1.2-1.3 2-3 2-4.9 0-3.9-3.1-7-7-7S1 4.1 1 8c0 1.9.8 3.6 2 4.9zM8 16c-4.4 0-8-3.6-8-8s3.6-8 8-8 8 3.6 8 8-3.6 8-8 8z"
                      />
                      <path
                        d="M8 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 1C6.3 9 5 7.7 5 6s1.3-3 3-3 3 1.3 3 3-1.3 3-3 3z"
                      />
                    </svg>
                    condition1
                    <svg
                      alt=""
                      aria-label=""
                      className="icon"
                      fill="#37a900"
                      fillRule="evenodd"
                      height="16"
                      name="icon--checkmark--solid"
                      role="img"
                      style={undefined}
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <title>
                        
                      </title>
                      <path
                        d="M8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16zm3.293-11.332L6.75 9.21 4.707 7.168 3.293 8.582 6.75 12.04l5.957-5.957-1.414-1.414z"
                      />
                    </svg>
                  </div>
                  <div
                    className="valueGroupItem"
                  >
                    <svg
                      alt=""
                      aria-label=""
                      className="icon"
                      fill="#6c7b85"
                      fillRule="evenodd"
                      height="16"
                      name="icon--user"
                      role="img"
                      style={undefined}
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <title>
                        
                      </title>
                      <path
                        d="M4 13.7c1.1.8 2.5 1.3 4 1.3s2.9-.5 4-1.3v-1.2c0-.8-.7-1.5-1.5-1.5h-5c-.8 0-1.5.7-1.5 1.5v1.2zm-1-.8v-.4C3 11.1 4.1 10 5.5 10h5c1.4 0 2.5 1.1 2.5 2.5v.4c1.2-1.3 2-3 2-4.9 0-3.9-3.1-7-7-7S1 4.1 1 8c0 1.9.8 3.6 2 4.9zM8 16c-4.4 0-8-3.6-8-8s3.6-8 8-8 8 3.6 8 8-3.6 8-8 8z"
                      />
                      <path
                        d="M8 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 1C6.3 9 5 7.7 5 6s1.3-3 3-3 3 1.3 3 3-1.3 3-3 3z"
                      />
                    </svg>
                    condition2
                    <svg
                      alt=""
                      aria-label=""
                      className="icon"
                      fill="#6c7b85"
                      fillRule="evenodd"
                      height="16"
                      name="icon--error--glyph"
                      role="img"
                      style={undefined}
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <title>
                        
                      </title>
                      <path
                        d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zM5.1 13.3L3.5 12 11 2.6l1.5 1.2-7.4 9.5z"
                      />
                    </svg>
                  </div>
                  <div
                    className="valueGroupItem"
                  >
                    <svg
                      alt=""
                      aria-label=""
                      className="icon"
                      fill="#6c7b85"
                      fillRule="evenodd"
                      height="16"
                      name="icon--user"
                      role="img"
                      style={undefined}
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <title>
                        
                      </title>
                      <path
                        d="M4 13.7c1.1.8 2.5 1.3 4 1.3s2.9-.5 4-1.3v-1.2c0-.8-.7-1.5-1.5-1.5h-5c-.8 0-1.5.7-1.5 1.5v1.2zm-1-.8v-.4C3 11.1 4.1 10 5.5 10h5c1.4 0 2.5 1.1 2.5 2.5v.4c1.2-1.3 2-3 2-4.9 0-3.9-3.1-7-7-7S1 4.1 1 8c0 1.9.8 3.6 2 4.9zM8 16c-4.4 0-8-3.6-8-8s3.6-8 8-8 8 3.6 8 8-3.6 8-8 8z"
                      />
                      <path
                        d="M8 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 1C6.3 9 5 7.7 5 6s1.3-3 3-3 3 1.3 3 3-1.3 3-3 3z"
                      />
                    </svg>
                    condition3
                    <svg
                      alt=""
                      aria-label=""
                      className="icon"
                      fill="#37a900"
                      fillRule="evenodd"
                      height="16"
                      name="icon--checkmark--solid"
                      role="img"
                      style={undefined}
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <title>
                        
                      </title>
                      <path
                        d="M8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16zm3.293-11.332L6.75 9.21 4.707 7.168 3.293 8.582 6.75 12.04l5.957-5.957-1.414-1.414z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="deploymentStatusClass"
          >
            <div
              className="innerContentBox"
            >
              <div
                className="subHeader"
              >
                <div
                  className="deploymentStatusHeader"
                >
                  Resource status
                </div>
                <span
                  className="statusTagFailed"
                >
                  failed
                </span>
              </div>
              <div
                className="deploymentProgress"
              >
                
              </div>
              <div
                className="deploymentStatusTime"
              >
                Deployed at 2019-07-05T09:50:56Z 
                Failed at 2019-07-05T10:20:56Z
              </div>
              <div>
                d1
                d2
                d3
                d4
              </div>
            </div>
          </div>
          <div
            className="targetClustersClass"
          >
            <div
              className="innerContentBox"
            >
              <div
                className="subHeader"
              >
                <div
                  className="deploymentStatusHeader"
                >
                  Target clusters
                </div>
                <div>
                  c1
                  c2
                  c3
                  c4
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="bx--modal-footer"
      >
        <div
          className="bx--modal__buttons-container"
        >
          <button
            className="bx--btn bx--btn--secondary"
            disabled={false}
            onClick={[Function]}
            tabIndex={0}
            type="button"
          />
          <button
            className="bx--btn bx--btn--primary"
            disabled={false}
            inputref={
              Object {
                "current": null,
              }
            }
            onClick={[Function]}
            tabIndex={0}
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
